corehq/motech/dhis2/static/dhis2/js/json_syntax_parse.js

```diff
@@ -0,0 +1,93 @@
+hqDefine('dhis2/js/json_syntax_parse', [
+], function () {
+    /*
+
+    This is a rewrite of the core of json-parse-even-better-errors library,
+    adapted to suit the needs of DHIS2 config handling.
+    Original can be found here: https://github.com/npm/json-parse-even-better-errors/blob/latest/index.js
+
+    */
+
+    var parseError = function (error, text, context, editor) {
+        if (!text) {
+            return {
+                message: error.message + ' while parsing empty string',
+                position: 0,
+            };
+        }
+        var badToken = error.message.startsWith('Unexpected token') ?
+            error.message.match(/^Unexpected token (.|\n) .*position\s+(\d+)/i)
+            : error.message.match(/^Unexpected ([^\s]+) .*position\s+(\d+)/i);
+
+        var errorIndex = badToken ? +badToken[2]
+            : error.message.match(/^Unexpected end of JSON.*/i) ? text.length - 1
+                : null;
+
+        var errorMsg = badToken ? error.message.replace(/^Unexpected token ./, `Unexpected token ${
+            JSON.stringify(badToken[1])
+        }`)
+            : error.message;
```

Comment:
Again, maybe personal preference, maybe Python bias, but this looks like it should either be an `if` clause, or the `error.message.replace()` call should be done first.

Reply:
Changed both places you pointed out to be if clauses for readability!

Reply:
Thank you.